### ISSUE TITLE
npm install fails if composer modules are in dependencies vs dev-dependencies

### ISSUE THREAD

**Issue by rddill-IBM** [hyperledger/composer#2324]
This issue has just appeared in the last few days. I have the composer modules referenced in the dependencies section of my package.json file: 
```
  "dependencies": {
    "body-parser": "^1.18.1",
    "cfenv": "^1.0.4",
    "composer-admin": "",
    "composer-cli": "",
    "composer-client": "",
    "composer-common": "",
    "composer-connector-embedded": "",
   ...
```
In the latest release, with these modules referenced in the normal dependencies location, the following error occurs: 
```
Creating Business Network Archive


Looking for package.json of Business Network Definition
	Input directory: /Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08
ParseException: Expected "-->", "@", "default", "o", "optional", "regex", "}", comment, end of line or whitespace but "v" found. Line 26 column 18
    at new ModelFile (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/introspect/modelfile.js:70:23)
    at ModelManager.addModelFiles (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/modelmanager.js:232:29)
    at Function._processModelFiles (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/businessnetworkdefinition.js:457:43)
    at Promise.resolve.then (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/businessnetworkdefinition.js:620:18)
    at process._tickCallback (internal/process/next_tick.js:109:7)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:502:3
{ ParseException: Expected "-->", "@", "default", "o", "optional", "regex", "}", comment, end of line or whitespace but "v" found. Line 26 column 18
    at new ModelFile (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/introspect/modelfile.js:70:23)
    at ModelManager.addModelFiles (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/modelmanager.js:232:29)
    at Function._processModelFiles (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/businessnetworkdefinition.js:457:43)
    at Promise.resolve.then (/Users/robertdill/Documents/GitHub/Z2B_Master/Chapter08/node_modules/composer-common/lib/businessnetworkdefinition.js:620:18)
    at process._tickCallback (internal/process/next_tick.js:109:7)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:502:3
  name: 'ParseException',
  fileLocation: 
   { start: { offset: 419, line: 26, column: 18 },
     end: { offset: 419, line: 26, column: 18 } },
  shortMessage: 'Expected "-->", "@", "default", "o", "optional", "regex", "}", comment, end of line or whitespace but "v" found.' }
```
The error disappears if the modules are referenced in the dev-dependencies section. This is a problem in that these modules need to be available for production, not just dev. I am unable to identify which file is creating this error. 

This can be seen in Chapter 9 in the Zero To Blockchain tutorial, located here: https://github.com/rddill-IBM/ZeroToBlockchain

**Comment by andyrichardson** [comment]
I can confirm this is an issue I have experienced also.

**Comment by davidkel** [comment]
There is a possibility that this issue might be the cause
https://github.com/hyperledger/composer/issues/3430
there is no plan at this time to fix it in 0.16.x release, but has been fixed in the next release, so be worth seeing if you get the same problem using the next release rather than latest release.

**Comment by davidkel** [comment]
So I believe this is fixed in master (and the next published releases) but is an issue in 0.16.x (latest published releases)

**Comment by davidkel** [comment]
Closing as this is believed to be fixed in the 0.18.x composer releases. If the 0.18.x and higher releases still have a problem then please raise a new issue.
